languages/rust/oso/src/host/class.rs

```diff
@@ -160,13 +178,17 @@ impl<T> Class<T> {
     }
 
     pub fn init(&self, fields: Vec<Term>, host: &mut Host) -> crate::Result<Instance> {
-        let instance = self.constructor.invoke(fields, host)?;
-        Ok(Instance {
-            name: self.name.clone(),
-            instance,
-            attributes: Arc::new(self.attributes.clone()),
-            methods: Arc::new(self.instance_methods.clone()),
-        })
+        if let Some(constructor) = &self.constructor {
+            let instance = constructor.invoke(fields, host)?;
+            Ok(Instance {
+                name: self.name.clone(),
+                instance,
+                attributes: Arc::new(self.attributes.clone()),
+                methods: Arc::new(self.instance_methods.clone()),
+            })
+        } else {
+            panic!("Invalid class, should already have a constructor")
```

Comment:
Classes don't _have_ to be constructable, right? No constructor seems okay, should be a regular error.

Reply:
Oh true, cool.